src/components/transfers/EditForm.tsx

```diff
@@ -0,0 +1,299 @@
+  const queryClient = useQueryClient()
+
+  const [sourceCampaignId, setSourceCampaignId] = useState(transfer.sourceCampaignId)
+  const [targetCampaignId, setTargetCampaignId] = useState(transfer.targetCampaignId)
+  const [sourceVaultId, setSourceVaultId] = useState(transfer.sourceVaultId)
+  const [targetVaultId, setTargetVaultId] = useState(transfer.targetVaultId)
+
+  const sourceVaults = vaults.filter((v) => v.campaignId === sourceCampaignId)
+  const targetVaults = vaults.filter((v) => v.campaignId === targetCampaignId)
+
+  const initialValues: TransferInput = {
+    status: transfer.status,
+    currency: transfer.currency,
+    amount: transfer.amount,
+    reason: transfer.reason,
+    documentId: transfer.documentId || '',
+    targetDate: dateParser(transfer.targetDate) || '',
+    approvedById: transfer.approvedById || '',
+    sourceCampaignId: sourceCampaignId,
+    sourceVaultId: sourceVaultId,
+    targetCampaignId: targetCampaignId,
+    targetVaultId: targetVaultId,
+  }
+
+  const mutation = useMutation<
+    AxiosResponse<TransferResponse>,
+    AxiosError<ApiErrors>,
+    TransferInput
+  >({
+    mutationFn: useEditTransfer(id),
+    onError: () => AlertStore.show(t('alerts:error'), 'error'),
+    onSuccess: () => {
+      AlertStore.show(t('alerts:create'), 'success')
+      queryClient.invalidateQueries(endpoints.transfer.viewTransfer(id).url)
+      router.push(routes.admin.transfer.index)
+    },
+  })
+
+  function handleSubmit(values: TransferInput) {
+    const data: TransferInput = {
+      status: values.status,
+      currency: values.currency,
+      amount: values.amount,
+      reason: values.reason,
+      documentId: values.documentId ? values.documentId : null,
+      targetDate: values.targetDate ? new Date(values.targetDate) : null,
+      approvedById: values.approvedById ? values.approvedById : null,
+      sourceCampaignId: sourceCampaignId,
+      sourceVaultId: sourceVaultId,
+      targetCampaignId: targetCampaignId,
+      targetVaultId: targetVaultId,
+    }
+    if (
+      !data.sourceCampaignId ||
+      !data.targetCampaignId ||
+      !data.sourceVaultId ||
+      !data.targetVaultId
+    ) {
+      return
+    }
+    mutation.mutate(data)
+  }
+
+  return (
+    <GenericForm
+      onSubmit={handleSubmit}
+      initialValues={initialValues}
+      validationSchema={validationSchema}>
+      <Box>
+        <Typography variant="h5" component="h2" sx={{ marginBottom: 2, textAlign: 'center' }}>
+          {t('edit-form-heading')}
+        </Typography>
+        <Grid container spacing={2} sx={{ width: 600, margin: '0 auto' }}>
+          <Grid item xs={12}>
+            <FormSelect name="status" label={t('status')}>
+              {(Object.values(TransferStatus) || []).map((status) => {
+                return (
+                  <MenuItem key={status} value={status}>
+                    {status}
+                  </MenuItem>
+                )
+              })}
+            </FormSelect>
```

Comment:
Maybe we need dedicated `TransferStatusSelect` component here instead of making it inline